dts/riscv/gigadevice/gd32vf103.dtsi

```diff
@@ -65,6 +65,36 @@
 			};
 		};
 
+		usart0: serial@40013800 {
+			compatible = "gd,gd32-usart";
+			reg = <0x40013800 0x400>;
+			interrupts = <37 0>;
```

Comment:
Please, review interrupt numbers. It starts from 56, 57, 58 by datasheet.
Test using hello_world but enable Shell interface, which forces use of interrupts. Then type something...

```
*** Booting Zephyr OS build v2.7.99-2195-g70074cb3f5bf  ***
Hello World! longan_nano

uart:~$ device list 
devices:
- UART_0 (READY)
uart:~$ 
```

Reply:
I've re-veryfy shell worked on each USART0, USART1, USART2.
(My original PR was correctly implments as your pointing out...
 It was tainted while rebasing...😨)